tower-buffer/src/service.rs

```diff
@@ -42,35 +41,30 @@ where
         T::Error: Send + Sync,
         Request: Send + 'static,
     {
-        Self::with_executor(service, bound, &mut DefaultExecutor::current())
+        let (tx, rx) = mpsc::channel(bound);
+        let (handle, worker) = Worker::new(service, rx);
+        tokio::spawn(worker);
+        Buffer { tx, handle }
```

Comment:
Should this just call `pair` directly and then `spawn`?

Reply:
Yes probably :) we could use a bit more clean up before we release again.